File: project-{topic}-advanced/src/database/entities/Visualization.ts
```typescript
import { Entity, PrimaryGeneratedColumn, Column, ManyToOne, CreateDateColumn, UpdateDateColumn } from 'typeorm';
import { Dataset } from './Dataset';
import { Dashboard } from './Dashboard';

export enum ChartType {
  BAR = 'bar',
  LINE = 'line',
  PIE = 'pie',
  SCATTER = 'scatter',
  AREA = 'area',
  TABLE = 'table',
  GAUGE = 'gauge',
}

@Entity('visualizations')
export class Visualization {
  @PrimaryGeneratedColumn('uuid')
  id!: string;

  @Column()
  name!: string;

  @Column({ type: 'enum', enum: ChartType })
  chartType!: ChartType;

  @Column('jsonb')
  configuration!: Record<string, any>; // E.g., { xField: 'sales', yField: 'month', seriesField: 'category' }

  @Column('jsonb', { nullable: true })
  dataMapping!: Record<string, any>; // Map dataset fields to chart properties

  @Column({ nullable: true })
  description!: string;

  @ManyToOne(() => Dataset, dataset => dataset.visualizations)
  dataset!: Dataset;

  @Column({ name: 'dataset_id' })
  datasetId!: string;

  @ManyToOne(() => Dashboard, dashboard => dashboard.visualizations, { nullable: true })
  dashboard!: Dashboard | null;

  @Column({ name: 'dashboard_id', nullable: true })
  dashboardId!: string | null;

  @CreateDateColumn({ name: 'created_at' })
  createdAt!: Date;

  @UpdateDateColumn({ name: 'updated_at' })
  updatedAt!: Date;
}
```

File: project-{topic}-complete/app/models/visualization.py
```python
import enum

from sqlalchemy import Column, Enum, ForeignKey, String, Text, JSON
from sqlalchemy.orm import relationship

from app.db.base import Base, BaseMixin


class ChartType(str, enum.Enum):
    """
    Enum for supported chart types.
    """
    BAR_CHART = "bar"
    LINE_CHART = "line"
    PIE_CHART = "pie"
    SCATTER_PLOT = "scatter"
    TABLE = "table"
    # Add more chart types (e.g., area, histogram, heatmap)


class Visualization(Base, BaseMixin):
    """
    SQLAlchemy model for Visualization.
    Represents a single chart or data display.
    """
    name = Column(String, index=True, nullable=False)
    description = Column(Text, nullable=True)
    chart_type = Column(Enum(ChartType), nullable=False)
    config = Column(JSON, nullable=False)  # JSON for chart configuration (e.g., axes, colors, filters)
    dataset_id = Column(ForeignKey("datasets.id"), nullable=False)
    owner_id = Column(ForeignKey("users.id"), nullable=False)

    # Relationships
    dataset = relationship("Dataset", back_populates="visualizations")
    owner = relationship("User", back_populates="visualizations")
    # No direct relationship to dashboard, but dashboards reference visualizations by ID (via JSON)

    def __repr__(self):
        return f"<Visualization {self.name} ({self.chart_type})>"
```